src/factories/digi/PulseDigi_factory.h

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2025 Minho Kim
+
+#pragma once
+
+#include "algorithms/digi/PulseDigi.h"
+#include "services/algorithms_init/AlgorithmsInit_service.h"
+#include "extensions/jana/JOmniFactory.h"
+
+namespace eicrecon {
+
+class PulseDigi_factory : public JOmniFactory<PulseDigi_factory, PulseDigiConfig> {
+
+public:
+  using AlgoT = eicrecon::PulseDigi;
+
+private:
+  std::unique_ptr<AlgoT> m_algo;
+
+  PodioInput<edm4eic::SimPulse> m_pulse_input{this};
+  PodioOutput<edm4eic::RawHGCROCHit> m_digi_output{this};
+
+  ParameterRef<double> m_time_window{this, "time_window", config().time_window};
+  ParameterRef<double> m_adc_phase{this, "adc_phase", config().adc_phase};
+  ParameterRef<double> m_toa_thres{this, "toa_thres", config().toa_thres};
+  ParameterRef<double> m_tot_thres{this, "tot_thres", config().tot_thres};
+  ParameterRef<unsigned int> m_capADC{this, "capADC", config().capADC};
+  ParameterRef<double> m_dyRangeADC{this, "dyRangeADC", config().dyRangeADC};
+  ParameterRef<unsigned int> m_capTOA{this, "capTOA", config().capTOA};
+  ParameterRef<double> m_dyRangeTOA{this, "dyRangeTOA", config().dyRangeTOA};
+  ParameterRef<unsigned int> m_capTOT{this, "capTOT", config().capTOT};
+  ParameterRef<double> m_dyRangeTOT{this, "dyRangeTOT", config().dyRangeTOT};
```

Comment:
```suggestion
  ParameterRef<double> m_time_window{this, "timeWindow", config().time_window};
  ParameterRef<double> m_adc_phase{this, "adcPhase", config().adc_phase};
  ParameterRef<double> m_toa_thres{this, "toaThres", config().toa_thres};
  ParameterRef<double> m_tot_thres{this, "totThres", config().tot_thres};
  ParameterRef<unsigned int> m_capADC{this, "capADC", config().capADC};
  ParameterRef<double> m_dyRangeADC{this, "dyRangeADC", config().dyRangeADC};
  ParameterRef<unsigned int> m_capTOA{this, "capTOA", config().capTOA};
  ParameterRef<double> m_dyRangeTOA{this, "dyRangeTOA", config().dyRangeTOA};
  ParameterRef<unsigned int> m_capTOT{this, "capTOT", config().capTOT};
  ParameterRef<double> m_dyRangeTOT{this, "dyRangeTOT", config().dyRangeTOT};
```
camelCase for JANA2  parameter names, and underscores for config names. Bonus points for making variable names consistent with what is used elsewhere.

Reply:
Thank you for pointing out that the JANA2 parameters were not using camelCase. I've applied your suggested changes in `PulseDigi_factory.h`. 
For the parameter names in `PulseDigiConfig.h`, would it be better to use underscores consistently for all of them? Since `CalorimeterHitDigiConfig` uses camelCase for its parameters (it has `capADC` and `dyRangeADC`), it is a bit hard for me to decided, so I wanted to follow your suggestion.